src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -2273,9 +2273,82 @@ def reindex(
         else:
             return self._reindex_axis_1(labels=labels, **kwargs)
 
+    def is_monotonic_decreasing(self) -> "SnowflakeQueryCompiler":
```

Comment:
why not directly return bool?

Reply:
The upstream modin frontend expects a query compiler object. We could override this, but I think for now it's best to keep as aligned with upstream as possible and revisit it in OSS modin.

Reply:
This sounds like a bug to me. We should override this IMO.

Reply:
I'm personally not certain that we *should* override this, since we want the QC to be returning QCs from methods in general, but if we do decide to override this, I think it should be done on upstream modin first, as otherwise, we'd have to move this method to `series_override.py`.